client/src/use/useSettings.ts

```diff
@@ -0,0 +1,61 @@
+import {
+  reactive, ref,
+} from '@vue/composition-api';
+
+export interface NewTrackSettings {
+    mode: string;
+    type: string;
+    modeSettings: {
+      Track: {
+        autoAdvanceFrame: boolean;
+      };
+      Detection: {
+        continuous: boolean;
+      };
+    };
+  }
+
+export default function useSettings() {
+  const newTrackSettings = reactive({
+    mode: ref('Track'),
+    type: ref('unknown'),
+    modeSettings: {
+      Track: {
+        autoAdvanceFrame: ref(false),
+      },
+      Detection: {
+        continuous: ref(true),
+      },
+    },
+  });
+
+  function saveSettings() {
+    //Get all settings
+    const Settings = {
+      newTrackSettings,
+    };
+    localStorage.setItem('Settings', JSON.stringify(Settings));
+  }
+
+  function updateNewTrackSettings(updatedNewTrackSettings: NewTrackSettings) {
```

Comment:
I'm fine with having different update functions for different functional parts of the settings.